### ISSUE TITLE
Rule Change: Extend table-column-count to flag rows with fewer cells than header

### ISSUE THREAD

**Issue by TKDev7** [eslint/markdown#422]
### What rule do you want to change?

table-column-count

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```markdown
| Header | Header | Header |
| ------ | ------ | ------ |
| Cell   | Cell   |        |  <!-- Currently VALID: fewer cells than header -->
```

### What does the rule currently do for this code?

The row with fewer cells is considered valid.

### What will the rule do after it's changed?

The rule will flag rows with fewer cells than the header.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

This change mirrors Markdownlint's `MD056` rule behavior, which flags both excess and missing cells. As for the option's name, I suggest `checkMissingCells`.

**Comment by lumirlumir** [comment]
This rule was originally implemented to check for excessive cells, as they can cause data loss.

Reference: https://github.com/eslint/markdown/pull/392#discussion_r2107332751

However, adding an option makes sense to me. @eslint/eslint-team, thoughts?

**Comment by nzakas** [comment]
I think it makes sense to add an option (off by default) for this. :+1: marking as accepted.
